### PR TITLE
[feat] Update etcd-backup flavor to use LinodeObjectStorageKey

### PR DESCRIPTION
**Pull request by bcm820**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: templates/addons/etcd-backup-restore/linode-obj.yaml
Comment (unnatiagg) [inline templates/addons/etcd-backup-restore/linode-obj.yaml:1]: should we set the region: ${LINODE_REGION} like this for simplicity? for both in kind: LinodeObjectStorageKey and kind: LinodeObjectStorageBucket

Comment (bcm820) [reply templates/addons/etcd-backup-restore/linode-obj.yaml:1]: Isn't `${OBJ_BUCKET_REGION:=${LINODE_REGION}}` using parameter expansion to allow for an override if `OBJ_BUCKET_REGION` is set, with `LINODE_REGION` as a fallback?

Comment (unnatiagg) [reply templates/addons/etcd-backup-restore/linode-obj.yaml:1]: OBJ_BUCKET_REGION  previously meant, clusterID which was of the form regionID-ordinal. Please refer the last file in my [PR](https://github.com/linode/cluster-api-provider-linode/commit/bd77f69300f96bd473f6b8aad642067794ea6fbc#diff-ec84f2f157d6e582be43faead06c60a3733c0bd33425375f4c8408195f82241d). Going forward we can either explicitly say that OBJ_BUCKET_REGION does not contain any ordinals anymore or we can directly set region: ${LINODE_REGION}.

Comment (bcm820) [reply templates/addons/etcd-backup-restore/linode-obj.yaml:1]: Looks like we just need to update our docs: https://github.com/linode/cluster-api-provider-linode/blob/main/docs/src/topics/etcd.md?plain=1#L37

**Review by eljohnson92 — APPROVED** [review]

**Review by unnatiagg — APPROVED** [review]

**Pull request merged**